### ISSUE TITLE
Problem with generate_sentence_pair_batch with a single positive example

### ISSUE THREAD

**Issue by jeanchilger** [pmbaumgartner/setfit#2]
Hi, first of all thanks for this amazing package.

I'm having some problem with the `generate_sentence_pair_batch` function. I'm working with a incremental training that starts with a single positive example. Turns out that, for this particular case, the function get stuck in a infinite loop on [line 46](https://github.com/pmbaumgartner/setfit/blob/f3a926361aac81e7da521c2e6d87dd53b3a2fd92/src/setfit/core.py#L46):

```python
    for current_sentence, current_label in zip(sentences, labels):
        # Choose itself? Seems wrong.
        positive_pair = random.choice(sent_lookup[current_label])
        while positive_pair == current_sentence: <-- here
            positive_pair = random.choice(sent_lookup[current_label])
```

Adding something like `len(sent_lookup[current_label]) > 1` might solve the infinite loop problem, but as you pointed in the comment of line 44, choosing itself might not be correct.

Why is not correct choosing itself and is there any solution (besides increasing the number of positive samples) I could apply to my scenario?

**Comment by pmbaumgartner** [comment]
Thanks for the issue! I had never thought about this incremental use case and it's quite interesting.

> Why is not correct choosing itself and is there any solution (besides increasing the number of positive samples) I could apply to my scenario?

This is a decision I made (not from the original `setfit` notebook) because the point of this task is to find pairs of (different) sentences with semantically similar information -- i.e. the same label. If you encountered the case where all the sentences were the same thing (i.e. they were paired with themselves), I don't think a model would generalize to understand that all of the things with the same label represent the same idea. I haven't tested this out myself, so it's totally possible there's no issue with this and I was a little too protective here, but that was my thinking.

As for a solution, you could just duplicate the input training data when the number of examples per label is one. For example, if you just have `sentences = ["this is a positive sentence", "this is a negative sentence"]` as your sentences and `labels = [1, 0]` as your labels, you could fake "doubling" your data with `sentences *= 2` and `labels *= 2`. This works because this function isn't checking whether the string literals passed as `sentences` are actually the same, just that the same item from the list of `sentences` isn't picked to pair with itself.

**Comment by pmbaumgartner** [comment]
Closed by #6
